packages/taquito-michelson-encoder/src/schema/storage.ts

```diff
@@ -34,22 +34,23 @@ function collapse(val: Token['val'] | any[], prim: string = PairToken.prim): Tok
       prim
     );
   }
-  if (val.prim === prim && val.args && val.args.length > 2) {
+  const extended = val as MichelsonV1ExpressionExtended
+  if (extended.prim === prim && extended.args && extended.args.length > 2) {
     return {
-      ...val,
+      ...extended,
       args: [
-        val.args?.[0],
+        extended.args?.[0],
         {
           prim: prim,
-          args: val.args?.slice(1),
+          args: extended.args?.slice(1),
         },
       ],
     };
   }
-  return val;
+  return extended;
 }
 
-function deepEqual(a: Token['val'] | any[], b: Token['val'] | any[]): boolean {
+function deepEqual(a: MichelsonV1Expression | MichelsonV1Expression[], b: MichelsonV1Expression | MichelsonV1Expression[]): boolean {
```

Comment:
I am wondering if `| MichelsonV1Expression[]` is necessary as it is already part of `MichelsonV1Expression`